glean-core/src/metrics/datetime.rs

```diff
@@ -0,0 +1,121 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#![allow(clippy::too_many_arguments)]
+
+use crate::metrics::time_unit::TimeUnit;
+use crate::metrics::Metric;
+use crate::metrics::MetricType;
+use crate::storage::StorageManager;
+use crate::util::get_iso_time_string;
+use crate::CommonMetricData;
+use crate::Glean;
+
+use chrono::{DateTime, FixedOffset, TimeZone};
+
+#[derive(Debug)]
+pub struct DatetimeMetric {
+    meta: CommonMetricData,
+    time_unit: TimeUnit,
+}
+
+impl MetricType for DatetimeMetric {
+    fn with_meta(meta: CommonMetricData) -> Self {
+        Self {
+            meta,
+            time_unit: TimeUnit::Day,
+        } // FIXME: How do we handle this?
+    }
+
+    fn meta(&self) -> &CommonMetricData {
+        &self.meta
+    }
+}
+
+impl DatetimeMetric {
+    pub fn new(meta: CommonMetricData, time_unit: TimeUnit) -> Self {
+        Self { meta, time_unit }
+    }
+
+    /// Public facing API for setting the metric to a date/time which
+    /// includes the timezone offset.
+    ///
+    /// Arguments:
```

Comment:
Below and in other places we tend to use a second-level header for this:
```suggestion
    /// ## Arguments:
```